n_ua.c

```diff
@@ -75,10 +117,10 @@ __attribute__((weak)) J *NoteUserAgent()
     char *compiler = (char *) ("iar" PLUS " " STRINGIFY(__VER__));
 #elif defined(__clang__)
     char *compiler = (char *) ("clang" PLUS " " __VERSION__);
-#elif defined(__GNUC__)
-    char *compiler = (char *) ("gcc" PLUS " " __VERSION__);
 #elif defined(__ATOLLIC__) && defined(__GNUC__)
     char *compiler = (char *) ("atollic gcc" PLUS " " __VERSION__);
+#elif defined(__GNUC__)
```

Comment:
This change is fine, but I will take this opportunity to belly ache that I would like to see modifications performed on the underlying library as separate PRs from unit-tests.

To me it seems cleaner, and maintains the separation of church and state.

Reply:
I'm fine with that going forward.